### PR TITLE
PARQUET-1942: Bump Arrow to 14.0.1

### PR DESCRIPTION
**Pull request by Fokko**
Make sure you have checked _all_ steps below.

This bumps Arrow many versions:

- Some API's have changed around the field.
- Add support for converting a map and also added a test.

### Jira

- [ ] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references them in the PR title. For example, "PARQUET-1234: My Parquet PR"
  - https://issues.apache.org/jira/browse/PARQUET-1942
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Comment by wgtmac** [comment]
It seems that we need to update the exclusion of japicmp

**Review by wgtmac — COMMENTED** [review]
Thanks for the update! I've left some comments.

**Comment by mapleFU** [comment]
Note that CVE in arrow, should we wait for 14.0.1 ? ( https://lists.apache.org/thread/whojbb1nlyfk8zdjrkzd8ozt0l1hrz57 )

**Comment by Fokko** [comment]
@mapleFU Thanks for pointing out. I think we can bump it to 14.0.1 right away 👍

File: parquet-arrow/pom.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
   <url>https://parquet.apache.org</url>
 
   <properties>
-    <arrow.version>0.10.0</arrow.version>
+    <arrow.version>14.0.0</arrow.version>
```
Comment (Fokko) [inline parquet-arrow/pom.xml:36]:
```suggestion
    <arrow.version>14.0.1</arrow.version>
```

**Review by wgtmac — APPROVED** [review]

**Comment by Fokko** [comment]
Thanks for the review @wgtmac

**Pull request merged**